atomics/include/desul/atomics/Lock_Array_Cuda.hpp

```diff
@@ -138,33 +138,36 @@ namespace {
 static int lock_array_copied = 0;
 inline int eliminate_warning_for_lock_array() { return lock_array_copied; }
 }  // namespace
+
+#ifdef __CUDACC_RDC__
+inline
+#else
+static
+#endif
+    void
+    copy_cuda_lock_arrays_to_device() {
+  if (::desul::Impl::lock_array_copied == 0) {
+    cudaMemcpyToSymbol(::desul::Impl::CUDA_SPACE_ATOMIC_LOCKS_DEVICE,
+                       &::desul::Impl::CUDA_SPACE_ATOMIC_LOCKS_DEVICE_h,
+                       sizeof(int32_t*));
+    cudaMemcpyToSymbol(::desul::Impl::CUDA_SPACE_ATOMIC_LOCKS_NODE,
+                       &::desul::Impl::CUDA_SPACE_ATOMIC_LOCKS_NODE_h,
+                       sizeof(int32_t*));
+  }
+  ::desul::Impl::lock_array_copied = 1;
+}
```

Comment:
Why the indirection?

Reply:
By that I mean why do you bother defining `copy_cuda_lock_arrays_to_device`

Reply:
`copy_cuda_lock_arrays_to_device` is also used in `init_lock_arrays_cuda` and `finalize_lock_arrays_cuda`, see https://github.com/desul/desul/blob/7a85900031e382616f84d8e35ff58be5e7ded504/atomics/src/Lock_Array_CUDA.cpp#L59-L90 and is also used for the rdc case.

Reply:
I am still confused why we have both but your changes are ok